File: 2018-01/err/open.c
#include <stdio.h>
#include <errno.h>

int main() {
	FILE *f = fopen("/etc/passwd", "r");
	if (f == NULL) {
		// handle error
		fprintf(stderr, "File opening failed: %d\n", errno);
		return 1;
	}
	// ...use the file...

	fclose(f);
	return 0;
}

File: 2018-01/err/open.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, err := os.Open("/etc/passwd")
	if err != nil {
		// handle error
		fmt.Fprintf(os.Stderr, "File opening failed: %s\n", err)
		os.Exit(1)
	}
	defer f.Close()

	// ...use the file...
}
